File: test_app/apps/mobile-expo/modules/scrollable-preview/ios/ScrollablePreviewModule.swift
import ExpoModulesCore

public class ScrollablePreviewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ScrollablePreview")

    // ── Trigger view (wraps child content, shows preview on long press) ──
    View(ScrollablePreviewTriggerView.self) {
      Prop("previewWidth") { (view: ScrollablePreviewTriggerView, value: CGFloat?) in
        view.previewWidth = value
      }
      Prop("previewHeight") { (view: ScrollablePreviewTriggerView, value: CGFloat?) in
        view.previewHeight = value
      }
      Events("onPrimaryAction", "onPreviewOpen", "onPreviewClose")
    }

    // ── Preview content container (the scrollable content inside the popup) ──
    View(ScrollablePreviewContentView.self) {}

    // ── Action item ──
    View(ScrollablePreviewActionView.self) {
      Prop("title") { (view: ScrollablePreviewActionView, value: String?) in
        view.actionTitle = value
      }
      Prop("icon") { (view: ScrollablePreviewActionView, value: String?) in
        view.iconName = value
      }
      Prop("destructive") { (view: ScrollablePreviewActionView, value: Bool?) in
        view.isDestructive = value ?? false
      }
      Events("onActionPress")
    }
  }
}

File: test_app/apps/mobile-expo/modules/scrollable-preview/android/src/main/java/expo/modules/scrollablepreview/ScrollablePreviewModule.kt
package expo.modules.scrollablepreview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ScrollablePreviewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ScrollablePreview")

    // ── Trigger view (wraps row content, opens the peek dialog on long press) ──
    View(ScrollablePreviewTriggerView::class) {
      Prop("previewWidth") { view: ScrollablePreviewTriggerView, value: Float? ->
        view.previewWidth = value
      }
      Prop("previewHeight") { view: ScrollablePreviewTriggerView, value: Float? ->
        view.previewHeight = value
      }
      Prop("previewHeightFraction") { view: ScrollablePreviewTriggerView, value: Float? ->
        view.previewHeightFraction = value
      }
      Events("onPrimaryAction", "onPreviewOpen", "onPreviewClose")
    }

    // ── Preview content container (the scrollable content inside the popup) ──
    View(ScrollablePreviewContentView::class) {}

    // ── Action item ──
    View(ScrollablePreviewActionView::class) {
      Prop("title") { view: ScrollablePreviewActionView, value: String? ->
        view.actionTitle = value
      }
      Prop("icon") { view: ScrollablePreviewActionView, value: String? ->
        view.iconName = value
      }
      Prop("destructive") { view: ScrollablePreviewActionView, value: Boolean? ->
        view.isDestructive = value ?: false
      }
      Events("onActionPress")
    }
  }
}
